db/friend.go

```diff
@@ -6,33 +6,33 @@ import (
 )
 
 type Friend struct {
-	DisplayName, MAC, IP string
+	DisplayName, MAC, IP, Username string
 }
 
 // Checks if the two users are friends given friend's display name
 func AreFriendsName(username string, friendDisplayName string) bool {
-	query := fmt.Sprintf("SELECT * FROM %s WHERE username=\"%s\" AND friend_display_name=\"%s\"", friendsTableName, username, friendDisplayName)
+	query := fmt.Sprintf("SELECT friend_display_name, friend_mac_address, friend_ip_address, friend_username FROM %s WHERE username=\"%s\" AND friend_display_name=\"%s\"", friendsTableName, username, friendDisplayName)
 	friends := ExecuteFriendsQuery(query)
 	return len(friends) > 0
 }
 
 // Checks if the two users are friends given friend's MAC Address
 func AreFriendsMac(username string, friendMacAddress string) bool {
```

Comment:
We shouldn't need both `AreFriendsName` and `AreFriendsMac` - just the first one should be sufficient.